pkg/datasource.go

```diff
@@ -162,29 +170,74 @@ func (o *OCIDatasource) QueryData(ctx context.Context, req *backend.QueryDataReq
 }
 
 func (o *OCIDatasource) testResponse(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	return &backend.QueryDataResponse{}, nil
-	// var ts GrafanaCommonRequest
-
-	// query := req.Queries[0]
-	// if err := json.Unmarshal(query.JSON, &ts); err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-
-	// //o.logger.Error("ts.Com is " + ts.Compartment)
-	// listMetrics := monitoring.ListMetricsRequest{
-	// 	CompartmentId: common.String(ts.Compartment),
-	// }
-	// reg := common.StringToRegion(ts.Region)
-	// o.metricsClient.SetRegion(string(reg))
-	// res, err := o.metricsClient.ListMetrics(ctx, listMetrics)
-	// if err != nil {
-	// 	return &backend.QueryDataResponse{}, err
-	// }
-	// status := res.RawResponse.StatusCode
-	// if status >= 200 && status < 300 {
-	// 	return &backend.QueryDataResponse{}, nil
-	// }
-	// return nil, errors.Wrap(err, fmt.Sprintf("list metrircs failed %s %d", spew.Sdump(res), status))
+
+	// GrafanaCommonRequest - captures the common parts of the search and metricsRequests
+	// the structure is used here to retrieve the TenancyOCID and CompartmentOCID
+	var ts GrafanaCommonRequest
+	query0 := req.Queries[0]
+
+	if err := json.Unmarshal(query0.JSON, &ts); err != nil {
+		return &backend.QueryDataResponse{}, err
+	}
+	o.logger.Debug("Testing OCI logs datasource", "TenancyOCID/CompartmentOCID", ts.TenancyOCID+"/"+ts.Compartment)
```

Comment:
The tenancy field in the data source configuration editor is actually very misleading since it actual is never used as part of the configuration of the data source or as part of the data source test. If the Environment dropdown is set to  'OCI Instance' then the tenancy will be the tenancy of the compute instance on which Grafana is running. If the Environment dropdown is set to  'local' then the tenancy will be the tenancy OCID referenced in the OCI configuration file to which Grafana has access.

So I don't know whether we should address removing this field from the config editor or not as part of this PR. If you'd prefer to not remove this field as part of this PR then I can write a new Jira issue to capture this work.

Reply:
Or we just leave it until the multi-tenancy feature is addressed since no one has noticed to this point.

Reply:
I think this change is beyond the scope of this particular PR honestly.

Reply:
I agree, I'll create a ticket to capture this item

Reply:
Nvm in looking at the code that configuration field is used by parts of the plugin, just not the datasource test logic.